### ISSUE TITLE
[data grid] Allow to customize how the `Intl.Collator` instance is created for sorting operators

### ISSUE THREAD

**Issue by JiiKooko** [mui/mui-x#5049]
### Duplicates

- [X] I have searched the existing issues

### Latest version

- [X] I have tested the latest version

### Current behavior 😯

Strings starting with character "Ä" are sorted like strings starting with "A". Same with letters "Å and Ö", they are sorted as "A and O"
In the linked sandbox "Ällö" should be sorted last when sorting ascending.

### Expected behavior 🤔

Sort works in alphabetical order.

### Steps to reproduce 🕹

Open https://codesandbox.io/s/data-grid-community-forked-2rzr4h
Sort ascending/descending
Check the position of "Ällö"

### Your environment 🌎

Check linked sandbox

**Comment by m4theushw** [comment]
Is "Ä" a complete different letter than "A" + accent in Scandinavian languages? Internally, we use the `Intl.Collator` API to sort the strings and it gives the following values:

![image](https://user-images.githubusercontent.com/42154031/171063633-c84f83ab-8517-44f1-9c1f-64e9521b9dad.png)

According to it, the sort you see is correct. Note that you can create a custom sort operator and pass custom options to it: https://mui.com/x/react-data-grid/sorting/#custom-comparator

**Comment by JiiKooko** [comment]
Yes "A" and "Ä" are complete different letters. Alphabet sort order  A,B,C...X,Y,Z,Å,Ä,Ö.

If I understood correctly Intl.Collator supports sorting using locales and would solve the issue: https://developer.mozilla.org/en-US/docs/Web/JavaScript/Reference/Global_Objects/Intl/Collator#using_locales

Is it possible to change data grid implementation to use locale based sorting and fallback to English or to not locale based? Implementing locale based sorting would help users working with accented characters and save time when there is no need to implement custom-comparators.

**Comment by laitilari** [comment]
Having the same problem. @JiiKooko Did you solve it?

**Comment by genepaul** [comment]
Has there been any movement on this? Is there any workaround? I suppose the only workaround is to create our own custom `sortComparator`, but it seems this should be more built-in to the component.

**Comment by michelengelen** [comment]
Thanks for the report, and to everyone who added the locale context. The default comparator still uses a plain `Intl.Collator()` and a global collator option never got built; the proposal above was not picked up and traction stayed low since. The supported approach is a locale-aware comparator on the affected columns:

```tsx
const collator = new Intl.Collator('sv');
const columns = [
  { field: 'name', sortComparator: (v1, v2) => collator.compare(String(v1), String(v2)) },
];
```

See https://mui.com/x/react-data-grid/sorting/#custom-comparator.

Closing as not planned. If demand picks up, fresh use cases and upvotes here can reopen the discussion.
